runtime/lua/vim/lsp/util.lua

```diff
@@ -1361,6 +1362,45 @@ local position_sort = sort_by_key(function(v)
   return {v.start.line, v.start.character}
 end)
 
+-- Gets the zero-indexed line from the given uri.
+-- For non-file uris, we load the buffer and get the line
+-- If a loaded buffer exists, then that is used.
+-- Otherwise we get the line using libuv which is a lot faster than loading the buffer.
+--@param uri string uri of the resource to get the line from
+--@param row number zero-indexed line number
+--@return string the line at row in filename
+function M.get_line(uri, row)
+  -- load the buffer if this is not a file uri
+  -- NOTE: when does this happen?
```

Comment:
```suggestion
  -- Custom language server protocol extension can result in servers sending URIs with custom schemes. Plugins are able to load these via `BufReadCmd` autocmds.
```

Reply:
Awesome. Will change it once I'm back home. Also about the col thing, just to be sure, I should revert that change for now right?

Reply:
> Also about the col thing, just to be sure, I should revert that change for now right?

I think it is okay to keep it as it is right now in the PR. `character_offset` would do another `nvim_buf_get_lines` call and as you correctly pointed out the `position.character` is already utf-16 and `character_offset` does the wrong thing. So although using `position.character` isn't the right fix, it doesn't break anything either.

Reply:
Sounds good. Thanks. Once home I'll amend the commit with your comment change.